cpp/src/slice2js/Gen.cpp

```diff
@@ -548,56 +532,43 @@ Slice::JsVisitor::splitComment(const ContainedPtr& p)
 }
 
 void
-Slice::JsVisitor::writeDocComment(const ContainedPtr& p, const string& deprecateReason, const string& extraParam)
+Slice::JsVisitor::writeDocCommentFor(const ContainedPtr& p)
 {
     StringList lines = splitComment(p);
-    if (lines.empty())
+    bool isDeprecated = p->isDeprecated(false);
+
+    if (lines.empty() && !isDeprecated)
     {
-        if (!deprecateReason.empty())
-        {
-            _out << nl << "/**";
-            _out << nl << " * @deprecated " << deprecateReason;
-            _out << nl << " **/";
-        }
+        // There's nothing to write for this doc-comment.
         return;
     }
 
     _out << nl << "/**";
 
-    bool doneExtraParam = false;
-    for (StringList::const_iterator i = lines.begin(); i != lines.end(); ++i)
+    for (const auto& line : lines)
     {
         //
         // @param must precede @return, so emit any extra parameter
         // when @return is seen.
         //
-        if (i->find("@return") != string::npos && !extraParam.empty())
-        {
-            _out << nl << " * " << extraParam;
-            doneExtraParam = true;
-        }
-        if ((*i).empty())
+        if (line.empty())
         {
             _out << nl << " *";
         }
         else
         {
-            _out << nl << " * " << *i;
+            _out << nl << " * " << line;
         }
     }
 
-    if (!doneExtraParam && !extraParam.empty())
+    if (isDeprecated)
     {
-        //
-        // Above code doesn't emit the comment for the extra parameter
-        // if the operation returns a void or doesn't have an @return.
-        //
-        _out << nl << " * " << extraParam;
-    }
-
-    if (!deprecateReason.empty())
-    {
-        _out << nl << " * @deprecated " << deprecateReason;
+        _out << nl << " * @deprecated";
+        if (auto reason = p->getDeprecationReason(false))
```

Comment:
why not true?

Reply:
My guess is that if a parent is deprecated, we only want to generate a comment once (on that parent),
instead of generating it on everything inside of it?

This is just what we were already doing though. If you look at the code underneath this:
`writeDocComment(p, getDeprecateReason(p, 0, "type"));`
The `0` means: don't bother with the parent.
And we _always_ pass `0` for javascript.